src/mscorlib/shared/System/IO/UnmanagedMemoryStream.cs

```diff
@@ -583,17 +604,40 @@ public override void Write(byte[] buffer, int offset, int count)
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNum);
             if (buffer.Length - offset < count)
                 throw new ArgumentException(SR.Argument_InvalidOffLen);
-            Contract.EndContractBlock();  // Keep contract validation in sync with WriteAsync(..)
 
-            if (!_isOpen) throw Error.GetStreamIsClosed();
-            if (!CanWrite) throw Error.GetWriteNotSupported();
+            WriteCore(new Span<byte>(buffer, offset, count));
+        }
+
+        public override void Write(ReadOnlySpan<byte> source)
+        {
+            if (GetType() == typeof(UnmanagedMemoryStream))
+            {
+                WriteCore(source);
+            }
+            else
+            {
+                // UnmanagedMemoryStream is not sealed, and a derived type may have overridden Write(byte[], int, int) prior
```

Comment:
BTW: There are a few internal streams in CoreLib that inherit from MemoryStream and UnmanagedMemoryStream. Is your thinking that we implement more efficient version of these overloads for all CoreLib/CoreFX streams, or just for the ones that matter (e.g. public Streams)?